src/Disks/IO/CachedOnDiskWriteBufferFromFile.cpp

```diff
@@ -225,10 +231,6 @@ void CachedOnDiskWriteBufferFromFile::nextImpl()
 
         throw;
     }
-
-    /// Write data to cache.
-    cacheData(working_buffer.begin(), size, throw_on_error_from_cache);
-    current_download_offset += size;
```

Comment:
why was this changed?

Reply:
This is test proved change. Actually the is just a bug which has not been triggered before but triggered now.

CachedOnDiskWriteBufferFromFile writes the data from working_buffer to the cache. It is exactly the buffer WriteBufferFromS3 allocates and maintains. The is no any guranty that the data still remain in the working_buffer after next is called. Just none. So the only right way to read from working_buffer in CachedOnDiskWriteBufferFromFile is to read from it before then working_buffer is passed to WriteBufferFromS3 by calling next. Ksenia and I discussed that changes in person.